client/src/components/Form/SignupForm.js

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import Form from "./Form";
+import { Button, TextField, Typography } from "@material-ui/core";
+import * as yup from "yup";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useAuth } from "../../hooks/useAuth";
+
+const schema = yup.object().shape({
+  username: yup
+    .string()
+    .required("A username is required")
+    .min(3, "Usernames must be between 3-20 characters in length")
+    .max(20, "Usernames must be between 3-20 characters in length")
+    .trim()
+    .matches(/^[_a-zA-Z0-9]*$/, {
+      message: "cannot contain special characters or spaces",
+    }),
+  email: yup
+    .string()
+    .required("An email is required")
+    .email("Must be a valid email"),
+  password: yup.string().required("Password is required").min(6).max(40),
+  password2: yup
+    .string()
+    .required("Please re-enter password")
+    .oneOf([yup.ref("password"), null], "Passwords must match"),
+});
+
+export default function SignupForm({ classes }) {
+  const { register, handleSubmit, errors } = useForm({
+    resolver: yupResolver(schema),
+  });
+
+  const auth = useAuth();
+
+  const onSubmit = (data) => {
+    const response = auth.signup(data);
+  };
+
+  return (
+    <Form handleSubmit={handleSubmit(onSubmit)}>
+      <Typography variant="h1">Create an account.</Typography>
+      <TextField
+        name="username"
+        label="Username"
+        className={classes.input}
+        inputRef={register}
+        error={errors.username ? true : null}
+        helperText={errors.username ? errors.username.message : null}
```

Comment:
These could be simplified to:
```suggestion
        error={!!errors.username}
        helperText={errors.username?.message}
```

Reply:
Oh thats so awesome! Added